sw/device/tests/keymgr_sideload_otbn_test.c

```diff
@@ -50,51 +50,26 @@ static const dif_otbn_err_bits_t kErrBitsOk = 0x0;
  * Runs the OTBN X25519 application.
  *
  * The X25519 app and sideloaded key should already be loaded into OTBN before
- * this routine is called.
+ * this routine is called. Causes CHECK-fail if the OTBN error code is not as
+ * expected.
  *
- * @param otbn_ctx OTBN context object
+ * @param otbn OTBN context object
  * @param[out] result Resulting Montgomery u-coordinate.
  * @param expect_err_bits Error code expected from OTBN ERR register.
- * @return true if the execution is successful (either successful completion
- *         or early exit through an EXPECTED error), otherwise false in case of
  * an unexpected error.
  */
-static bool run_x25519_app(otbn_t *otbn_ctx, uint32_t *result,
+static void run_x25519_app(dif_otbn_t *otbn, uint32_t *result,
                            dif_otbn_err_bits_t expect_err_bits) {
   // Copy the input argument (Montgomery u-coordinate).
-  CHECK(otbn_copy_data_to_otbn(otbn_ctx, sizeof(kEncodedU), &kEncodedU,
-                               kOtbnVarEncU) == kOtbnOk);
+  otbn_testutils_write_data(otbn, sizeof(kEncodedU), &kEncodedU, kOtbnVarEncU);
 
   // Run the OTBN program and wait for it to complete.
   LOG_INFO("Starting OTBN program...");
-  CHECK(otbn_execute(otbn_ctx) == kOtbnOk);
-  if (otbn_busy_wait_for_done(otbn_ctx) != kOtbnOk) {
-    // Error during execution; retrieve error bits and instruction count to
-    // help with debugging and then exit early.
-    dif_otbn_err_bits_t err_bits;
-    CHECK(dif_otbn_get_err_bits(&otbn_ctx->dif, &err_bits) == kDifOk);
-    uint32_t insn_count;
-    CHECK(dif_otbn_get_insn_cnt(&otbn_ctx->dif, &insn_count) == kDifOk);
-    LOG_INFO(
-        "OTBN encountered an error.\n  Error bits: 0x%08x\n  Instruction "
-        "count: 0x%08x",
-        err_bits, insn_count);
-    CHECK(err_bits == expect_err_bits);
-    return true;  // Exit early (return true as CHECK above passes).
-  }
-
-  // Check that an error was not expected from OTBN.
-  if (expect_err_bits == kOtbnInvalidKeyErr) {
-    LOG_ERROR("An error 0x%08x was expected with OTBN and was not encountered.",
-              expect_err_bits);
-    return false;
-  }
+  otbn_testutils_execute(otbn);
+  otbn_testutils_wait_for_done(otbn, expect_err_bits);
```

Comment:
Just a note on this part: @gdessouky's code here has basically been merged into `otbn_testutils_wait_for_done` , so **all** tests now print extra debugging information if OTBN encounters an unexpected error during execution.